core/console/commands/ThemeController.php

```diff
@@ -0,0 +1,130 @@
+                return $this->outputError('Abort by user.');
+            } else {
+                $themeName = $newName;
+            }
+        }
+    
+        $availableModules = implode(', ', array_column(Yii::$app->getFrontendModules(), 'id'));
+        $themeLocation = $this->prompt("Enter the theme location where to generate (as path alias e.g. app, $availableModules):", ['default' => 'app']);
+        $themeLocation = '@' . ltrim($themeLocation, '@');
+    
+        preg_match("#^@[A-z]+#", $themeLocation, $newThemeLocation);
+    
+        if ($newThemeLocation[0] !== $themeLocation) {
+            if (!$this->confirm("We have changed the name to '{$newThemeLocation}'. Do you want to proceed with this name?")) {
+                return $this->outputError('Abort by user.');
+            } else {
+                $themeLocation = $newThemeLocation;
+            }
+        }
+        
+        $basePath = $themeLocation . DIRECTORY_SEPARATOR . 'themes' . DIRECTORY_SEPARATOR . $themeName;
+        $themeFolder = Yii::getAlias($basePath);
+    
+        if (file_exists($themeFolder)) {
+            return $this->outputError("The folder " . $themeFolder . " exists already.");
+        }
+    
+        $this->outputInfo("Theme path alias: " . $basePath);
+        $this->outputInfo("Theme real path: " . $themeFolder);
+        if (!$this->confirm("Do you want continue?")) {
+            return $this->outputError('Abort by user.');
+        }
+        
+        $folders = [
+            '',
+            'resources',
+            'views',
+            'views/layouts',
+            'views/cmslayouts',
+        ];
+
+        foreach ($folders as $folder) {
+            FileHelper::createDirectory($themeFolder . DIRECTORY_SEPARATOR . $folder);
+        }
+        
+        $contents = [
+            $themeFolder. DIRECTORY_SEPARATOR . 'theme.json' => $this->renderJson($basePath, $themeName),
+        ];
+        
+        foreach ($contents as $fileName => $content) {
+            FileHelper::writeFile($fileName, $content);
+        }
+        
+        return $this->outputSuccess("Theme files has been created successfully. Please run `".$_SERVER['PHP_SELF']." import` to loaded into the database.");
+    }
+    
+    /**
+     * Render the json template.
+     *
+     * @param string $basePath
+     * @param string $themeName
+     * @return string
+     */
+    private function renderJson(string $basePath, string $themeName)
+    {
+        $themeConfig = new ThemeConfig($basePath, []);
+        $themeConfig->name = $themeName;
+        
+        if ($this->confirm('Inherit from other theme?')) {
+            $themeConfig->parentTheme = $this->prompt("Enter the base path (e.g. `@app/themes/blank`) of the parent theme:",
+                [
+                    'default' => '@app/themes/blank',
+                    'validator' => function ($input, &$error) {
```

Comment:
@nadar The validation for the `parentTheme` input is much harder to test. I dont think it make sense to solve this at this point.